core/Pos/Binary/Core/Coin.hs

```diff
@@ -115,20 +120,25 @@ hdrToParam h
     | isClear h 4 = (3, fromIntegral (h .&. 0x0f))
     | otherwise   = (4, fromIntegral (h .&. 0x0f))
 
-decodeVarint :: Peek Word64
-decodeVarint = do
-    (nbBytes, acc) <- hdrToParam <$> getWord8
-    conts <- replicateM nbBytes getWord8
-    let val = orAndShift acc conts
-    when (expectedContBytes val /= length conts) $ fail "not canonical encoding"
-    return val
+decodeMegaMicros :: [Word8] -> Decoder s (Word64, [Word8])
```

Comment:
As I didn't want to mess around too much with the existing format of the `Coin`, I have ported the old `Peek Coin` to the CBOR equivalent `Decoder s Coin`, and I had to create this `decodeMegaMicros` function which would act in lieu of the old `Peek Word64`. @dcoutts We should probably review this as the first item in our agenda, as I start to fear that this implementation is decoding what used to be a `VarInt` by looking for 1 or 2 consecutive `Word8` in memory, whereas our CBOR serialisation is totally different, so I'm wondering whether we might be generating garbage. The roundtrip tests on `Coin` definitely passes, but I guess an extra pair of eyes are never harmful!